src/routes/associations.js

```diff
@@ -20,32 +21,40 @@ module.exports = function (app, model, Implementation, integrator, opts) {
   }
 
   function getAssociation(request) {
-    return {
-      associationName:
-         _.first(_.last(request.route.path.split('/')).split('.csv'))
-    };
+    const pathSplit = request.route.path.split('/');
+    let associationName = pathSplit[pathSplit.length - 1];
+
+    if (nodePath.extname(associationName) === '.csv') {
+      associationName = nodePath.basename(associationName, '.csv');
+    } else if (associationName === 'count') {
+      associationName = pathSplit[pathSplit.length - 2];
+    }
+
+    return { associationName };
   }
 
-  function index(request, response, next) {
-    var association = getAssociation(request);
-    var params = _.extend(request.query, request.params, association);
-    var models = Implementation.getModels();
-    var associationField = getAssociationField(params.associationName);
-    var associationModel = _.find(models, function (model) {
+  function getContext(request) {
+    const association = getAssociation(request);
+    const params = _.extend(request.query, request.params, association);
+    const models = Implementation.getModels();
+    const associationField = getAssociationField(params.associationName);
+    const associationModel = _.find(models, function (model) {
       return Implementation.getModelName(model) === associationField;
     });
 
-    return new Implementation.HasManyGetter(model, associationModel, opts,
-      params)
+    return { params, associationModel };
+  }
+
+  function list(request, response, next) {
+    const { params, associationModel } = getContext(request);
+
+    return new Implementation.HasManyGetter(model, associationModel, opts, params)
       .perform()
       .then(function (results) {
         var records = results[0];
-        var count = results[1];
-        var fieldsSearched = results[2];
+        var fieldsSearched = results[1];
 
-        var meta = {
-          count: count
-        };
+        const meta = null;
```

Comment:
why don't you just send null in the function call?

Reply:
done 💪